deployment/src/main/java/io/quarkiverse/cxf/deployment/QuarkusCxfProcessor.java

```diff
@@ -212,93 +222,94 @@ public void build(
-                }
+            Collection<ClassInfo> implementors = index.getAllKnownImplementors(DotName.createSimple(sei));
 
-            } else {
+            if (implementors != null && !implementors.isEmpty()) {
+                factoryBean = factoryBean == null ? createQuarkusJaxWsServiceFactoryBean(sei, bus) : factoryBean;
 
                 for (ClassInfo wsClass : implementors) {
                     String implementor = wsClass.name().toString();
+                    String wsName = wsNameDefault;
                     if (implementor.contains(".")) {
                         wsName = implementor.substring(implementor.lastIndexOf('.') + 1);
                     } else {
                         wsName = implementor;
                     }
                     additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(implementor));
-                    AnnotationInstance bindingType = wsClass.classAnnotation(BINDING_TYPE_ANNOTATION);
-                    if (bindingType != null) {
-                        soapBinding = bindingType.value().asString();
-                    }
-                    cxfWebServices.produce(new CxfWebServiceBuildItem(cxfBuildTimeConfig.path, sei, soapBinding, wsNamespace,
-                            wsName, wrapperClassNames, implementor));
+                    String soapBinding = Optional.ofNullable(wsClass.classAnnotation(BINDING_TYPE_ANNOTATION))
+                            .map(bindingType -> bindingType.value().asString())
+                            .orElse(soapBindingDefault);
+                    cxfWebServices.produce(new CxfWebServiceBuildItem(cxfBuildTimeConfig.path, sei, soapBinding,
+                            wsNamespaceDefault, wsName, factoryBean.getWrappersClassNames(), implementor));
                 }
             }
-            cxfWebServices.produce(new CxfWebServiceBuildItem(cxfBuildTimeConfig.path, sei, soapBinding, wsNamespace,
-                    wsName, wrapperClassNames));
 
-            proxies.produce(new NativeImageProxyDefinitionBuildItem(wsClassInfo.name().toString(),
-                    "javax.xml.ws.BindingProvider", "java.io.Closeable", "org.apache.cxf.endpoint.Client"));
+            if (clientSEIsInUse.contains(sei)) {
+                factoryBean = factoryBean == null ? createQuarkusJaxWsServiceFactoryBean(sei, bus) : factoryBean;
+
+                AnnotationInstance webserviceClient = findWebServiceClientAnnotation(index, wsClassInfo.name());
+                String wsNamespace = wsNamespaceDefault;
+                String wsName = wsNameDefault;
+                if (webserviceClient != null) {
+                    wsName = webserviceClient.value("name").asString();
+                    wsNamespace = webserviceClient.value("targetNamespace").asString();
+                }
+                cxfWebServices.produce(new CxfWebServiceBuildItem(cxfBuildTimeConfig.path, sei, soapBindingDefault, wsNamespace,
```

Comment:
it is soapBinding and not soapBindingDefault I think no ?

Reply:
Each implementor can set a another `soapBinding`, that's why I introduced `soapBindingDefault` to avoid that one implementor influences another implementor or a client.

Reply:
PS: the is no adjustment for the client, that's why it uses `soapBindingDefault` directly

Reply:
Strange that cxf clients can not been on soap 1.2 maybe we need configuration for that part.  Or detect this annotation on sei and not only implementor

Reply:
Haven't you introduced a SOAP 1.2 config option for clients not so long ago? #209

Reply:
Lol I forget about it.... I am dory!

Reply:
dory 😄 

But I must say I'm a bit confused what that soap handling TODO in QuarkusCxfProcessor is about then.
Is it about a _global_ default that should be configurable?